src/node_modules/_file-explorer.js

```diff
@@ -0,0 +1,292 @@
+    if (event.type === 'mouseout') {
+      var exitedTo = event.toElement || event.relatedTarget
+      if (this.contains(exitedTo)) return
+      return this.removeChild(deleteButton)
+    }
+    this.appendChild(deleteButton)
+  }
+  function getPath (el) { return getPathRoot(container, el) }
+  function getElement (path) { return getElementRoot(container, path) }
+  function deletePath (event) {
+    event.cancelBubble = true
+    var el = this.parentElement.parentElement
+    var path = getPath(el)
+    if (confirm(`
+      Do you really want to delete "${path.file||path.folder}" ?
+      ${path.folder ? '(and all contained files and folders)' : ''}
+    `)) {
+      path = path.file || path.folder
+      el.parentElement.removeChild(el)
+      removeSubtree(files, path)
+    }
+  }
+  function editModeOn (event) {
+    var classes = this.parentElement.parentElement.parentElement.className
+    if (~classes.indexOf('hasFocus') && !this.getAttribute('contenteditable')) {
+      textUnderEdit = this.innerText
+      this.setAttribute('contenteditable', true)
+      this.classList.add(css.rename)
+      this.focus()
+    }
+  }
+  function editModeOff (event) {
+    if (event.type === 'blur' || event.which === 27 || event.which === 13) {
+      if (event.which === 13) {
+        event.preventDefault()
+        save = true
+      } else if (textUnderEdit !== this.innerText) var save = confirm(`
+        Do you want to rename?
+      `)
+      if (save) {
+        var newPath = getPath(this.parentElement.parentElement.parentElement)
+        newPath = newPath.folder || newPath.file
+        var oldPath = newPath.split('/')
+        oldPath.pop()
+        oldPath.push(textUnderEdit)
+        oldPath = oldPath.join('/')
+        var isReadOnly = updateAllFiles(oldPath, newPath)
+        if (isReadOnly) {
+          alert(`
+            couldn't rename - path contains "readonly" items
+          `)
+          this.innerText = textUnderEdit
+        } else textUnderEdit = this.innerText
+      }
+      else this.innerText = textUnderEdit
+      this.removeAttribute('contenteditable')
+      this.classList.remove(css.rename)
+    }
+  }
+  function updateAllFiles (oldPath, newPath) {
+    var allFiles = files.list()
+    var allPaths = Object.keys(allFiles)
+    for (var i=0, len=allPaths.length, isOK=true; i<len; i++) {
+      if (files.isReadOnly(allPaths[i])) {
+        isOK = false
+        break
+      }
+    }
+    if (isOK) return (allPaths.forEach(function (path) {
+      if (~path.indexOf(oldPath)) {
+        var newName = path.replace(oldPath, newPath)
+        files.rename(path, newName)
+      }
+    }),false)
+    return true
+  }
+}
+/******************************************************************************
+  HELPER FUNCTIONS
+******************************************************************************/
+function adaptEnvironment (el, focus, hover) {
+  var node1 = el.parentElement.parentElement.parentElement
```

Comment:
ok, i'll do better names. sry.
It's mainly because of the TreeView component from `remix` that i'm using and i need to patch it here and there and they don't have good names, but i can try to make up names and use at least their html tags